server/util/testing/flags/flags.go

```diff
@@ -2,51 +2,42 @@ package flags
 
 import (
 	"flag"
-	"reflect"
-	"strings"
+	"sync"
 	"testing"
 
 	"github.com/buildbuddy-io/buildbuddy/server/config"
-	"github.com/stretchr/testify/assert"
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
 	"github.com/stretchr/testify/require"
 )
 
-// Set a flag value and register a cleanup function to restore the flag
-// to its original value after the given test is complete.
-func Set(t testing.TB, name, value string) {
-	config.RegisterAndParseFlags()
-	f := flag.Lookup(name)
-	if f == nil {
-		t.Fatalf("Undefined flag: %s", name)
-	}
-	original := f.Value.String()
-	flag.Set(name, value)
-	wasSet := config.TestOnlySetFlag(name, true)
-	t.Cleanup(func() {
-		flag.Set(name, original)
-		config.TestOnlySetFlag(name, wasSet)
+var populateFlagsOnce sync.Once
+
+func PopulateFlagsFromData(t testing.TB, testConfigData []byte) {
+	populateFlagsOnce.Do(func() {
+		// add placeholder type for type adding by testing
+		flagutil.AddTestFlagTypeForTesting(flag.Lookup("test.benchtime").Value, struct{}{})
+		c, err := config.NewConfiguratorFromData(testConfigData)
+		c.ReconcileFlagsAndConfig()
+		require.NoError(t, err)
 	})
 }
 
-// CheckFlagsAgainstConfig checks that all defined flags containing a `.` are
-// present in the config struct.
-func CheckFlagsAgainstConfig(t *testing.T) {
-	configurator, err := config.NewConfiguratorFromData([]byte{})
+// Set a flag value and register a cleanup function to restore the flag
+// to its original value after the given test is complete.
+func Set(t testing.TB, name string, value interface{}) {
+	origValue, err := flagutil.DereferencedValueFromFlagName(name)
+	if status.IsNotFoundError(err) {
+		log.Warningf("Attempted to set %s to %v, but %s was not found.", name, value, name)
```

Comment:
If we make this a test failure instead of just logging, I think it would prevent these warnings from building up over time, and also prevent people from setting flags that don't exist and then a test passes even though it shouldn't. wdyt?

Reply:
I made this an error before and it made testenv sad, but I now think there's a better way to fix this: manually guard in testenv against setting a nonexistent flag. Not sure why I didn't just do that in the first place. Fixing this also revealed many places where we were changing options that are never referenced.